### PR TITLE
Add Datatypes examples to Spanner sample.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: spanner/test/spannerTest.php [hunk 1]
```diff
@@ -470,6 +470,111 @@ public function testUpdateDataWithBatchDML()
...
+    }
+
+    /**
+     * @depends testInsertDataWithDatatypes
+     */
+    public function testQueryDataWithBool()
+    {
+        $output = $this->runCommand('query-data-with-bool');
+        self::$lastUpdateDataTimestamp = time();
+        $this->assertContains('eVnueId: 19, VenueName: Venue 19, OutdoorVenue: True', $output);
```
Comment (bshaffer) [inline spanner/test/spannerTest.php]: TYPO! It's breaking the tests!

File: spanner/src/query_data_with_bool.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_bool($instanceId, $databaseId)
+{
+    $spanner = new SpannerClient();
+    $instance = $spanner->instance($instanceId);
+    $database = $instance->database($databaseId);
+
+    $exampleBool = True;
```
Comment (jdpedrie) [inline spanner/src/query_data_with_bool.php]:
```suggestion
    $exampleBool = true;
```

[PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#25-keywords-and-truefalsenull) requires that `true`, `false`, and `null` be lowercase.

File: spanner/src/insert_data_with_datatypes.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ */
+function insert_data_with_datatypes($instanceId, $databaseId)
+{
+    $spanner = new SpannerClient();
+    $instance = $spanner->instance($instanceId);
+    $database = $instance->database($databaseId);
+
+    $operation = $database->transaction(['singleUse' => true])
+        ->insertBatch('Venues', [
+            ['VenueId' => 4, 'VenueName' => 'Venue 4', 'VenueInfo' => base64_encode('Hello World 1'),
```
Comment (jdpedrie) [inline spanner/src/insert_data_with_datatypes.php]: These are hard to visually parse. Could they be formatted differently?

File: spanner/test/spannerTest.php (hunk 1 above)
Comment (jsimonweb) [reply spanner/test/spannerTest.php]: Fixed.

File: spanner/src/query_data_with_bool.php (hunk 2 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_bool.php]: Done.

File: spanner/src/insert_data_with_datatypes.php (hunk 3 above)
Comment (jsimonweb) [reply spanner/src/insert_data_with_datatypes.php]: Updated.

File: spanner/src/insert_data_with_datatypes.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            ['VenueId' => 42,
+            'VenueName' => 'Venue 42',
+            'VenueInfo' => base64_encode('Hello World 3'),
+            'Capacity' => 3000,
+            'AvailableDates' => ['2020-10-01', '2020-10-07'],
+            'LastContactDate' => '2018-10-01',
+            'OutdoorVenue' => false,
+            'PopularityScore' => 0.72598,
+            'LastUpdateTime' => $spanner->commitTimestamp()],
+        ])
```
Comment (bshaffer) [inline spanner/src/insert_data_with_datatypes.php]: Try this instead:

```php
$operation = $database->transaction(['singleUse' => true])
    ->insertBatch('Venues', [
        [
            'VenueId' => 4,
            'VenueName' => 'Venue 4',
            'VenueInfo' => base64_encode('Hello World 1'),
            'Capacity' => 1800,
            'AvailableDates' => ['2020-12-01', '2020-12-02', '2020-12-03'],
            'LastContactDate' => '2018-09-02',
            'OutdoorVenue' => false,
            'PopularityScore' => 0.85543,
            'LastUpdateTime' => $spanner->commitTimestamp()
        ], [
            'VenueId' => 19,
            'VenueName' => 'Venue 19',
            'VenueInfo' => base64_encode('Hello World 2'),
            'Capacity' => 6300,
            'AvailableDates' => ['2020-11-01', '2020-11-05', '2020-11-15'],
            'LastContactDate' => '2019-01-15',
            'OutdoorVenue' => true,
            'PopularityScore' => 0.98716,
            'LastUpdateTime' => $spanner->commitTimestamp()
        ], [
            'VenueId' => 42,
            'VenueName' => 'Venue 42',
            'VenueInfo' => base64_encode('Hello World 3'),
            'Capacity' => 3000,
            'AvailableDates' => ['2020-10-01', '2020-10-07'],
            'LastContactDate' => '2018-10-01',
            'OutdoorVenue' => false,
            'PopularityScore' => 0.72598,
            'LastUpdateTime' => $spanner->commitTimestamp()
        ],
    ])
    ->commit();
```

File: spanner/src/query_data_with_array.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ */
+function query_data_with_array($instanceId, $databaseId)
+{
+    $spanner = new SpannerClient();
+    $instance = $spanner->instance($instanceId);
+    $database = $instance->database($databaseId);
+
+    $exampleArray = [
+        new Date(new \DateTime('2020-10-01')),
+        new Date(new \DateTime('2020-11-01'))
```
Comment (bshaffer) [inline spanner/src/query_data_with_array.php]: why is this a future date?

File: spanner/src/query_data_with_array.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ * Queries sample data from the database using SQL with an ARRAY parameter.
+ * Example:
+ * ```
+ * query_data_with_array($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_array($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_array.php]: This should match the region tag: `function query_data_with_array_parameters`

File: spanner/src/query_data_with_bool.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+ * Queries sample data from the database using SQL with a BOOL parameter.
+ * Example:
+ * ```
+ * query_data_with_bool($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_bool($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_bool.php]: same, let's keep these consistent with region tags and add `_parameter`

File: spanner/src/query_data_with_bytes.php [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+ * Queries sample data from the database using SQL with a BYTES parameter.
+ * Example:
+ * ```
+ * query_data_with_bytes($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_bytes($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_bytes.php]: `query_data_with_bytes_parameter`

File: spanner/src/query_data_with_date.php [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+ * Queries sample data from the database using SQL with a DATE parameter.
+ * Example:
+ * ```
+ * query_data_with_date($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_date($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_date.php]: `query_data_with_date_parameter`

File: spanner/src/query_data_with_float.php [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+ * Queries sample data from the database using SQL with a FLOAT64 parameter.
+ * Example:
+ * ```
+ * query_data_with_float($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_float($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_float.php]: `query_data_with_float_parameter`

File: spanner/src/query_data_with_int.php [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+ * Queries sample data from the database using SQL with a INT64 parameter.
+ * Example:
+ * ```
+ * query_data_with_int($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_int($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_int.php]: `query_data_with_int_parameter`

File: spanner/src/query_data_with_string.php [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+ * Queries sample data from the database using SQL with a STRING parameter.
+ * Example:
+ * ```
+ * query_data_with_string($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_string($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_string.php]: `query_data_with_string_parameter`

File: spanner/src/query_data_with_timestamp.php [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+ * Queries sample data from the database using SQL with a TIMESTAMP parameter.
+ * Example:
+ * ```
+ * query_data_with_timestamp($instanceId, $databaseId);
+ * ```
+ *
+ * @param string $instanceId The Spanner instance ID.
+ * @param string $databaseId The Spanner database ID.
+ */
+function query_data_with_timestamp($instanceId, $databaseId)
```
Comment (bshaffer) [inline spanner/src/query_data_with_timestamp.php]: `query_data_with_timestamp_parameter`

File: spanner/src/insert_data_with_datatypes.php (hunk 4 above)
Comment (jsimonweb) [reply spanner/src/insert_data_with_datatypes.php]: Updated.

File: spanner/src/query_data_with_array.php (hunk 5 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_array.php]: Example checks for available dates at a venue. :)

File: spanner/src/query_data_with_array.php (hunk 6 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_array.php]: Done.

File: spanner/src/query_data_with_date.php (hunk 9 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_date.php]: Done.

File: spanner/src/query_data_with_float.php (hunk 10 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_float.php]: Done.

File: spanner/src/query_data_with_int.php (hunk 11 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_int.php]: Done.

File: spanner/src/query_data_with_string.php (hunk 12 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_string.php]: Done.

File: spanner/src/query_data_with_timestamp.php (hunk 13 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_timestamp.php]: Done.

File: spanner/src/query_data_with_bytes.php (hunk 8 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_bytes.php]: Done.

File: spanner/src/query_data_with_bool.php (hunk 7 above)
Comment (jsimonweb) [reply spanner/src/query_data_with_bool.php]: Done.

File: spanner/src/query_data_with_array.php (hunk 5 above)
Comment (bshaffer) [reply spanner/src/query_data_with_array.php]: Should we make this dynamic so it doesn't break next year? Something like:

```php
$exampleArray = [
      new Date(new \DateTime('+1 week')),
      new Date(new \DateTime('+1 week'))
];
```

If those dates are hardcoded in the documentation page though, this won't work

Comment (jsimonweb) [reply spanner/src/query_data_with_array.php]: Those dates are hardcoded in the 'insert_data_with_datatypes' function
so they will stop being future dates later next year but the 'query_data_with_array_parameter' function will continue to work as will the tests.

**Pull request merged**